### ISSUE TITLE
Oauth / AAD / Entra ID support for mssql

### ISSUE THREAD

**Issue by jonaslb** [sfu-db/connector-x#682]
We use OAuth / `AzureDefaultCredential` for authentication to mssql and I would like to use this with connectorx as well.

It looks like tiberius has support for this with the `AADToken` auth method: <https://docs.rs/tiberius/0.12.3/tiberius/enum.AuthMethod.html#variant.AADToken>. I haven't tested that it works, but I suppose this would basically accept the OAuth token that you get from `AzureDefaultCredential.token`.

There is, as far as I can see, currently no way to encode this token in the URL for connectorx. It could perhaps be a `token` parameter or it could be encoded in the user/password field somehow.

**Comment by keen85** [comment]
@jonaslb have you found a way to make this work?

**Comment by jonaslb** [comment]
No, never looked at it again. Now, Tiberius looks dead and Microsoft are working on their own rust driver for mssql, which might be preferred anyway: https://github.com/microsoft/mssql-rs
